pkgs/development/python-modules/taichi/default.nix

```diff
@@ -0,0 +1,250 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  autoPatchelfHook,
+  autoAddDriverRunpath,
+  llvmPackages_15,
+  cmake,
+  ninja,
+  setuptools,
+  wheel,
+  pybind11,
+  scikit-build,
+  git,
+  pythonImportsCheckHook,
+  pythonCatchConflictsHook,
+
+  colorama,
+  dill,
+  matplotlib,
+  numpy,
+  rich,
+
+  cudaPackages,
+  elfutils,
+  libbfd,
+  libdwarf,
+  libffi,
+  libGL,
+  libX11,
+  libXrandr,
+  libXinerama,
+  libXcursor,
+  libXi,
+  libxml2,
+  vulkan-headers,
+  vulkan-loader,
+
+  gtest,
+  pytest,
+
+  config,
+  cudaSupport ? config.cudaSupport,
+  rocmSupport ? config.rocmSupport,
+  stdenv,
+  enableMetal ? stdenv.isDarwin,
+  enableOpenGL ? stdenv.isLinux,
+  enableVulkan ? stdenv.isLinux,
+}:
+
+let
```

Comment:
use rec instead

Reply:
This is the only point I am not sure how to do.
Most attributes are needed both in the taichi cmake build phase (compiling the c-api) and the taichi python packaging in `buildPythonPackage`. If I used `rec`, I would have to maintain everything twice 🤔